cpp/src/dataframe/cudf_table.cuh

```diff
@@ -467,7 +467,7 @@ public:
     {
       col_type& my_value{static_cast<col_type*>(my_column)[my_row_index]};
       col_type const& other_value{static_cast<col_type const*>(other_column)[other_row_index]};
-      cudf::detail::unwrap(my_value) = cudf::detail::unwrap(other_value);
+      my_value = other_value;
```

Comment:
It is indeed assignment.

It is copying a single element from one type erased buffer to another type erased buffer (both of the same type).

Reply:
But has the other buffer been constructed, or just allocated? (Conceptually, I mean)

Reply:
The other buffer is allocated, and may or may not already be filled with values.

Reply:
Placement new -> copy construction
assignment -> assignment operator
movement / assignment of temporary -> move assignment operator

different code in theory, in our case - same thing. For strings it could be different, but then, frankly, the entire current architecture of cudf pretty much breaks down on (variable-length) strings.